.orchestra/support/mass-testing/check-regression

```diff
@@ -134,57 +142,68 @@ def main() -> int:
+            other_errors_count += 1
             continue
 
         limits = tests_limits[entry.test_name]
         if entry.status == "OOM":
             if not (limits.min_memory <= entry.previous_max_rss <= limits.max_memory):
-                errors.append(
-                    f"{entry.full_name()} exited due to excessive memory usage "
-                    + f"(used {entry.current_max_rss:.0f} bytes, "
-                    + f"was {entry.previous_max_rss:.0f} bytes)"
-                )
-                error_entries.append(entry)
+                timeout_oom_count += 1
+                errors.append(entry)
             continue
 
         if entry.status == "TIMED_OUT":
             if not (limits.min_timeout <= entry.previous_elapsed_time <= limits.max_timeout):
-                errors.append(
-                    f"{entry.full_name()} took too long to execute "
-                    + f"(took {entry.current_elapsed_time:.2f}s, "
-                    + f"was {entry.previous_elapsed_time:.2f}s)"
-                )
-                error_entries.append(entry)
+                timeout_oom_count += 1
+                errors.append(entry)
             continue
 
         # If here, the `status` string is unknown, error out
-        errors.append(f"{entry.full_name()} has unknown status: {entry.status}")
-        error_entries.append(entry)
+        errors.append(entry)
+        other_errors_count += 1
 
     if args.json is not None:
         with open(args.json, "w") as f:
-            json.dump([asdict(e) for e in error_entries], f)
+            json.dump([asdict(e) for e in errors], f)
 
-    if len(errors) > 0:
+    timeout_oom_over_limit = timeout_oom_count > (len(entries) * timeout_oom_fail_margin)
+    do_fail = other_errors_count > 0 or timeout_oom_over_limit
+
+    if do_fail:
         print("Regression failed, the following errors were encountered:")
-        for error in errors:
-            print(error)
-        return 1
-    else:
-        return 0
+    elif len(errors) > 0:
+        print("Regression passed, however the following skippable errors were encountered:")
+
+    for entry in errors:
+        if entry.status == "FAILED":
+            print(f"{entry.full_name()} exited with a non-zero exit code")
+        elif entry.status == "CRASHED":
+            print(f"{entry.full_name()} has crashed")
+        elif entry.status == "OOM":
+            print(
+                f"{entry.full_name()} exited due to excessive memory usage "
+                + f"(used {entry.current_max_rss:.0f} bytes, "
+                + f"was {entry.previous_max_rss:.0f} bytes)"
+            )
+        elif entry.status == "TIMED_OUT":
+            print(
+                f"{entry.full_name()} took too long to execute "
+                + f"(took {entry.current_elapsed_time:.2f}s, "
+                + f"was {entry.previous_elapsed_time:.2f}s)"
+            )
+        else:
+            print(f"{entry.full_name()} has unknown status: {entry.status}")
```

Comment:
Convert this script to print messages on `stderr`.

Reply:
Done